crates/sui-core/src/authority_active/execution_driver/mod.rs

```diff
@@ -110,17 +110,26 @@ where
         debug!(digest=?d, "Pending execution for certificate.");
 
         // Sync and Execute with local authority state
-        net.sync_certificate_to_authority_with_timeout_inner(
-            c.clone(),
-            active_authority.state.name,
-            &local_handler,
-            tokio::time::Duration::from_secs(10),
-            10,
-        )
-        .await?;
-
-        // Remove from the execution list
-        executed.push(*i);
+        match net
+            .sync_certificate_to_authority_with_timeout_inner(
+                c.clone(),
+                active_authority.state.name,
+                &local_handler,
+                tokio::time::Duration::from_secs(10),
+                10,
+            )
+            .await
+        {
+            Err(err) => {
```

Comment:
Good call -- I assumed here exec is infallible but clearly it is no (for example when we are offline).